`make otel-from-tree` results in ambiguous import errors

### Component(s)

_No response_

### What happened?

Running the following:

```shell
mkdir collector
cd collector
git clone git@github.com:open-telemetry/opentelemetry-collector.git
git clone git@github.com:open-telemetry/opentelemetry-collector-contrib.git
cd opentelemetry-collector-contrib
make otel-from-tree
make install-tools
make
```


Results in lots of ambiguous import errors:

```text
connector/countconnector/internal/metadata/generated_status.go:6:2: ambiguous import: found package go.opentelemetry.io/collector/component in multiple modules:
	go.opentelemetry.io/collector v0.91.0 (/Users/awest/github/collector/opentelemetry-collector/component)
	go.opentelemetry.io/collector/component v0.91.0 (/Users/awest/go/1.21.5/pkg/mod/go.opentelemetry.io/collector/component@v0.91.0)
internal/coreinternal/traceutil/traceutil.go:9:2: ambiguous import: found package go.opentelemetry.io/collector/pdata/pcommon in multiple modules:
	go.opentelemetry.io/collector v0.91.0 (/Users/awest/github/collector/opentelemetry-collector/pdata/pcommon)
	go.opentelemetry.io/collector/pdata v1.0.0 (/Users/awest/go/1.21.5/pkg/mod/go.opentelemetry.io/collector/pdata@v1.0.0/pcommon)
internal/coreinternal/traceutil/traceutil.go:10:2: ambiguous import: found package go.opentelemetry.io/collector/pdata/ptrace in multiple modules:
	go.opentelemetry.io/collector v0.91.0 (/Users/awest/github/collector/opentelemetry-collector/pdata/ptrace)
	go.opentelemetry.io/collector/pdata v1.0.0 (/Users/awest/go/1.21.5/pkg/mod/go.opentelemetry.io/collector/pdata@v1.0.0/ptrace)
internal/filter/filterconfig/config.go:11:2: ambiguous import: found package go.opentelemetry.io/collector/pdata/plog in multiple modules:
	go.opentelemetry.io/collector v0.91.0 (/Users/awest/github/collector/opentelemetry-collector/pdata/plog)
	go.opentelemetry.io/collector/pdata v1.0.0 (/Users/awest/go/1.21.5/pkg/mod/go.opentelemetry.io/collector/pdata@v1.0.0/plog)
...
```

### Collector version

v0.91.0

### Environment information

_No response_

### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

_No response_

--- a/Makefile
+++ b/Makefile
@@ -310,7 +310,7 @@ otel-from-tree:
 	# 2. Run `make otel-from-tree` (only need to run it once to remap go modules)
 	# 3. You can now build contrib and it will use your local otel core changes.
 	# 4. Before committing/pushing your contrib changes, undo by running `make otel-from-lib`.
-	$(MAKE) for-all CMD="$(GOCMD) mod edit -replace go.opentelemetry.io/collector=$(SRC_ROOT)/../opentelemetry-collector"
+	$(MAKE) for-all CMD="$(GOCMD) mod edit -replace go.opentelemetry.io/collector=$(SRC_PARENT_DIR)/opentelemetry-collector"
 
 .PHONY: otel-from-lib
 otel-from-lib:

--- a/Makefile.Common
+++ b/Makefile.Common
@@ -20,6 +20,8 @@ endif
 
 # SRC_ROOT is the top of the source tree.
 SRC_ROOT := $(shell git rev-parse --show-toplevel)
+# SRC_PARENT_DIR is the absolute path of source tree's parent directory
+SRC_PARENT_DIR :=  $(shell dirname $(SRC_ROOT))
 
 # build tags required by any component should be defined as an independent variables and later added to GO_BUILD_TAGS below
 GO_BUILD_TAGS=""

